libgit/repo.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libgit
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/keybase/kbfs/libfs"
+	"github.com/keybase/kbfs/libkbfs"
+	"github.com/pkg/errors"
+)
+
+const (
+	kbfsRepoDir    = ".kbfs_git"
+	kbfsConfigName = "kbfs_config"
+)
+
+// This character set is what Github supports in repo names.  It's
+// probably to avoid any problems when cloning onto filesystems that
+// have different Unicode decompression schemes
+// (https://en.wikipedia.org/wiki/Unicode_equivalence).  There's no
+// internal reason to be so restrictive, but it probably makes sense
+// to start off more restrictive and then relax things later as we
+// test.
+var repoNameRE = regexp.MustCompile(`^([a-zA-Z0-9][a-zA-Z0-9_.-]+)$`)
```

Comment:
Did you intend to have `+` instead of `*`? It seems this would disallow single character names. Also if you decide to change this, would you mind changing `.` into `\.` as well while you are there? I was confused a bit and thought that was a bug :)